sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -142,13 +194,77 @@ namespace Azure { namespace Core {
 
   public:
     /**
-     * @brief Constructs a new context with no deadline, and no value associated.
+     * @brief Constructs a context with no deadline, and no value associated.
      *
      */
     Context() : m_contextSharedState(std::make_shared<ContextSharedState>()) {}
 
     /**
-     * @brief Creates a context with a deadline.
+     * @brief Creates a context with a deadline when you do not have an existing Context object.
+     *
+     * @param deadline A point in time after which a context expires.
+     *
+     */
+    Context(DateTime const& deadline)
+        : m_contextSharedState(std::make_shared<ContextSharedState>(nullptr, deadline))
+    {
+    }
+
+    /**
+     * @brief Copies a context.
+     *
+     * This operation copies one context to another. Context objects are copied by reference,
+     * so the new context will share the same state as the original context. This also means
+     * that cancelling one context cancels all contexts which are copied from the original
+     * context.
+     *
+     * 	@param other Another context to copy.
+     *
+     */
+    Context(Context const&) = default;
```

Comment:
explicit?

Reply:
Similarly to the default constructor, how can this constructor be implicitly invoked?